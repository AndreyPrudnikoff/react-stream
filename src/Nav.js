import React from 'react';
import { Link } from "react-router-dom";

function Nav(props) {
    return (
        <nav className="navbar navbar-expand-lg navbar-light bg-primary">
            <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavDropdown"
                    aria-controls="navbarNavDropdown" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"/>
            </button>
            <div className="collapse navbar-collapse" id="navbarNavDropdown">
                <ul className="navbar-nav">
                    <li className="nav-item active">
                        <Link className="nav-link" to="/">Home</Link>
                    </li>
                    <li className="nav-item">
                        <Link className="nav-link" to="/list-users">Users</Link>
                    </li>
                    <li className="nav-item">
                        <Link className="nav-link" to="/videos">Videos</Link>
                    </li>
                </ul>
            </div>
        </nav>
    );
}

export default Nav;
