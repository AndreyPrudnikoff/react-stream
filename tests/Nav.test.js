import React from 'react'
import { cleanup, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Link, BrowserRouter as Router } from 'react-router-dom'
import Nav from '../src/Nav.js'

test('Links didn\'t change', () => {
    const component = renderer.create(
        <Router><Nav/></Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

afterEach(cleanup);

it('Navigation component exist', () => {
    const component = render(
        <Router><Nav/></Router>,
    );
});