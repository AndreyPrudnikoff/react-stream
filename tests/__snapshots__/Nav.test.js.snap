// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links didn't change 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-primary"
>
  <button
    aria-controls="navbarNavDropdown"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarNavDropdown"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarNavDropdown"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/list-users"
          onClick={[Function]}
        >
          Users
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/videos"
          onClick={[Function]}
        >
          Videos
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
